marl_classification/networks/ft_extractor.py

```diff
@@ -47,16 +47,15 @@ def __init__(self, f: int) -> None:
         self.__seq_conv = nn.Sequential(
             nn.Conv2d(1, 16, (3, 3), padding=1),
             nn.GELU(),
-            nn.MaxPool2d(2, 2),
             nn.BatchNorm2d(16),
             nn.Conv2d(16, 32, (3, 3), padding=1),
             nn.GELU(),
-            nn.MaxPool2d(2, 2),
             nn.BatchNorm2d(32),
+            nn.MaxPool2d(2, 2) if f > 2 else nn.Identity(),
             nn.Flatten(1, -1),
         )
 
-        self.__out_size = 32 * (f // 4) ** 2
+        self.__out_size = 32 * ((f // 2) ** 2 if f > 2 else f ** 2)
```

Comment:
any reason why we are introducing changes in the feature extractor?

Reply:
The code crashes if it receives a f value lower than 4 because the self.__out_size variable is set to 0. I changed the code so that the the self.__out_size is not set to 0 whenever the f value is lower than 4.